src/Voron/Data/Tables/TableValueCompressor.cs

```diff
@@ -241,7 +245,15 @@ private void MaybeTrainCompressionDictionary(Table table, FixedSizeTree etagsTre
                             // keep it, we can compress data with a dictionary that we rolled back, and end up with data
                             // corruption!!!
                             // ****************************************
-                            llt.Environment.CompressionDictionariesHolder.Remove(newId);
+                            bool removed = llt.Environment.CompressionDictionariesHolder.Remove(newId);
+
+                            if (Logger.IsInfoEnabled == false)
```

Comment:
how often we hit this? would like to avoid hitting this in a hot path too often

Reply:
As stated in the comment to the `Remove()` method, this is a *very* rare operation.